### PR TITLE
Added invalid code and code system conditions, and code coverage

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## Terminology Service PR

Jira Ticket: [MAT-4208](https://jira.cms.gov/browse/MAT-4208)
(Optional) Related Tickets:

### Summary

Reopening the ticket to fix a few scenarios as I clearly missed the failure to retrieve code from VSAC.

1. VSAC is returning back with 400 Bad requests if a code of code system is not found and the response from VSAC has its own error codes to identify the specific error. 
2. I tried to mention most of it in java docs or in comments
3. Updated test cases to match the new scenarios and added a code coverage report ( Jacoco ).
4. Not enough knowledge to figure out if we actually need "maven-surefire-plugin" for now it's commented out since it is causing me to generate a code coverage report. But please do correct me if there is a better way to configure it. 
5. Code coverage is still down to 74%. We need to improve it but not in the scope of this story.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by sb-cecilialiu — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -168,28 +168,47 @@
           </excludes>
         </configuration>
       </plugin>
+      <!--      <plugin>-->
+      <!--        <groupId>org.apache.maven.plugins</groupId>-->
+      <!--        <artifactId>maven-surefire-plugin</artifactId>-->
```
Comment (adongare) [inline pom.xml]: don't need surefire anymore?

Comment (RohitKandimalla) [reply pom.xml]: That is what I am unsure about, I know that it runs unit tests and generate reports, which are usually text and XML files, Are they useful for us in any way? 

I added the same comment in this PR summary, I personally believe that having a JACOCO plugin that could give us interactive HTML code coverage is very useful.

Comment (RohitKandimalla) [reply pom.xml]: that is why I just commented on it so that we can discuss and see what everyone thinks.

Comment (adongare) [reply pom.xml]: ah, I missed your description. I'm not sure either. Maybe our java gurus have some ideas Joe, Nate, Cicilia, or Greg

**Review by adongare — DISMISSED** [review]

**Review by adongare — APPROVED** [review]

**Pull request merged**